game-core/src/main/java/games/strategy/engine/framework/startup/mc/SetupPanelModel.java

```diff
@@ -119,39 +120,44 @@ public void login() {
               GameRunner.hideMainFrame();
               lobbyFrame.setVisible(true);
 
+              boolean passwordChanged = false;
               if (lobbyClient.isPasswordChangeRequired()) {
-                ChangePasswordPanel.newChangePasswordPanel()
-                    .show(lobbyFrame)
-                    .map(RsaAuthenticator::hashPasswordWithSalt)
-                    .ifPresentOrElse(
-                        pass -> {
-                          final String error = lobbyClient.updatePassword(pass);
-                          if (error == null) {
-                            DialogBuilder.builder()
-                                .parent(lobbyFrame)
-                                .title("Success")
-                                .infoMessage("Password successfully updated!")
-                                .showDialog();
-                          } else {
-                            DialogBuilder.builder()
-                                .parent(lobbyFrame)
-                                .title("Error")
-                                .errorMessage("Error updating password.\nError:" + error)
-                                .showDialog();
-                          }
-                        },
-                        () ->
-                            DialogBuilder.builder()
-                                .parent(lobbyFrame)
-                                .title("Password Not Updated")
-                                .errorMessage(
-                                    "Password not updated, you will need to request a new "
-                                        + "temporary password to log in again.")
-                                .showDialog());
+                try {
+                  passwordChanged =
```

Comment:
Your thoughts on putting a `do {} while(!passwordChanged);` loop here?
Without this loop people have to manually re-click a button to get the same dialog. Just a thought, popups can also by super annoying if users want to completely cancel the whole login procedure anyways

Reply:
Few thoughts:
- This pop-up is shown on login after using a temp password, so there is no button to be clicked. That does make me think the message text should be updated to tell the user to use the "account > change password" feature before they log out.
- If there are any persistent errors on reset password, the do-while loop would effectively become infinite.
- The recurring pop-up can be a bit annoying, instead it would be a better UX to disallow for the window to be closed or cancelled until the 'do' condition is fulfilled. In this case we could add a prompt like "are you really sure you want to close this window?"